projects/serializers.py

```diff
@@ -52,61 +54,55 @@ class Meta:
+    """Project serializer for the case where only basic project info is needed."""
+    class Meta:
+        model = Project
+        fields = ("id", "title", "acronym", )
+        read_only_fields = ("id", "title", "acronym", )
+
+
 class ProjectControlSerializer(serializers.ModelSerializer):
-    catalog_data = serializers.SerializerMethodField()
-    component_data = serializers.SerializerMethodField()
-
-    def get_catalog_data(self, obj):
-        """
-        Get the Catalog data for a given Control.
-        """
-        project_catalog = obj.catalog
-        control_id = self.context.get("control_id")
-        catalog = CatalogTools(project_catalog.file_name.path)
-        control_data = catalog.get_control_data_simplified(control_id)
+    control = ControlSerializer(read_only=True)
+    project = BasicViewProjectSerializer(read_only=True)
+    catalog_data = serializers.SerializerMethodField(read_only=True)
+    component_data = serializers.SerializerMethodField(read_only=True)
+
+    class Meta:
+        model = ProjectControl
+        fields = ("status", "project", "control", "catalog_data", "component_data", )
+
+    def get_catalog_data(self, obj: ProjectControl) -> dict:
+        """Get the Catalog data for a given Control."""
+        catalog = CatalogTools(obj.control.catalog.file_name.path)
+        control_data = catalog.get_control_data_simplified(control_id=self.context.get("control_id"))
         control_data["version"] = catalog.catalog_title
+
         return control_data
 
-    def get_component_data(self, obj):
-        """
-        Get the narratives from any Component that includes the given Control.
-        """
-        control_id = self.context.get("control_id")
-        components: dict = {}
-        if hasattr(obj, "components"):
-            components = self.get_control_data(obj.components, control_id)
-        return components
+    def get_component_data(self, obj: ProjectControl) -> dict:
+        """Get the narratives from any Component that includes the given Control."""
+        if obj.project.components.exists():
+            return self._get_control_data(obj.project.components.all(), self.context.get("control_id"))
 
-    class Meta:
-        model = Project
-        fields = (
-            "id",
-            "title",
-            "acronym",
-            "catalog_data",
-            "component_data",
-        )
+        return {}
 
-    def get_control_data(self, components, control_id):
+    @staticmethod
+    def _get_control_data(components: QuerySet, control_id: str) -> dict:
         component_data = {
             "responsibility": "Allocated",
             "components": {"inherited": {}, "private": {}},
         }
         count = 0
         responsibility: str = ""
-        for c in components.all():
-            status = "inherited" if c.status == 2 else "private"
-            if control_id in c.controls:
+        for component in components:
```

Comment:
Praise: I'm a huuuuuge fan of not using single letter variables.  I don't typically block PRs over it but I definitely approve the improvements 😉